sktime/forecasting/reservoir.py

```diff
@@ -0,0 +1,187 @@
+    >>> y = load_airline()
+    >>> forecaster = NextRC()
+    >>> forecaster.fit(y)
+    NextRC(...)
+    >>> y_pred = forecaster.predict(fh=[1,2,3])
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "handles-missing-data": True,
+        "scitype:y": "univariate",
+        "ignores-exogeneous-X": False,
+    }
+
+    def __init__(self, regressor=None):
+        self.regressor = regressor
+        self.X_ = None
+        self.y_ = None
+        self.delay_ = 1
+        self.d_linear_ = None
+        self.d_nonlinear_ = None
+        self.fitted_y = False
+        super(NextRC, self).__init__()
+
+    def _feature_vector(self, X, k=None):
+        if k is None:
+            k = self.delay_
+
+        (n_samples, n_features) = X.shape
+
+        # size of the linear part of the feature vector
+        self.d_linear_ = d_linear = k * n_features
+        # size of the non-linear part of feature vector
+        self.d_nonlinear_ = d_nonlinear = int(
+            d_linear * (d_linear + 1) * (d_linear + 2) / 6
+        )
+        # total size
+        d = d_linear + d_nonlinear
+
+        # create an array to hold the linear part of the feature vector
+        features = np.zeros((d_linear, n_samples))
+
+        # fill in the linear part of the complete feature vector
+        for delay in range(k):
+            for j in range(delay, n_samples):
+                features[n_features * delay : n_features * (delay + 1), j] = X[
+                    j - delay, :
+                ]
+
+        # create full feature vector
+        O_features = np.zeros((d, n_samples))
+
+        # copy over the linear part
+        O_features[0:d_linear, :] = features
+
+        # fill in the non-linear part
+        cnt = 0
+        for row in range(d_linear):
+            for column in range(row, d_linear):
+                for span in range(column, d_linear):
+                    O_features[d_linear + cnt] = (
+                        O_features[row, :] * O_features[column, :] * O_features[span, :]
+                    )
+                    cnt += 1
+
+        return O_features
+
+    def _fit(self, X, y=None, fh=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series with which to fit the forecaster.
+        X : pd.DataFrame, default=None
+            Exogenous variables are ignored
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        self.X_ = X
+        self.y_ = y
+        self.delay_ = 2
+
+        if self.regressor is None:
+            self.regressor_ = sklearn.linear_model.Ridge(
+                alpha=1.0e-3, fit_intercept=False
+            )
+
+        (n_samples, n_features) = X.shape
```

Comment:
Also `n_samples` has no use. We can ignore it. Right?

Reply:
Yep, you could write `(_, n_features) = X.shape` or `n_features = X.shape[1]` or just ignore it.